src/transformers/utils/hub.py

```diff
@@ -628,8 +623,58 @@ def _resumable_file_manager() -> "io.BufferedWriter":
     return cache_path
 
 
-def get_file_from_repo(
-    path_or_repo: Union[str, os.PathLike],
+def try_to_load_from_cache(cache_dir, repo_id, filename, revision=None):
```

Comment:
This is my solution for
> 3. a new function `try_to_load_from_cache` is created, which can definitely leave in Transformers forever if it's not deemed suitable for `huggingface_hub`.

I think this can leave in hf hub and be used there to return a cached file when internet/hf.co is down.

Reply:
would make sense to have in `huggingface_hub` (just not specific to models but would support any repo type)